### ISSUE TITLE
when Modal set to hide, the status bar will flicker

### ISSUE THREAD

**Issue by walkerXiong** [facebook/react-native#14087]
![e4inh](https://cloud.githubusercontent.com/assets/7894779/26290655/b2415c52-3edb-11e7-84bf-ef9f46f4c21f.gif)
here is code: and i want to know if it is a issue? or , whether it is a universal phenomenon.
```
'use strict';
import React, {Component} from 'react';
import {
    View,
    Text,
    StyleSheet,
    TouchableOpacity,
    Modal
} from 'react-native';

import NavActivity from '../components/NavActivity';

class TestPage extends Component {
    constructor(props) {
        super(props);
        this.state = {
            modalVisible: false
        }
    }

    _setModalVisible(param) {
        this.setState({modalVisible: param});
    }

    render() {
        return (
            <View style={Styles.wrap}>
                <NavActivity
                    navigator={this.props.navigator}
                    title={{
                        title: '测试页面'
                    }}/>
                <TouchableOpacity
                    onPress={() => this.setState({modalVisible: true})}>
                    <Text>show modal</Text>
                </TouchableOpacity>
                <Modal
                    animationType={'fade'}
                    transparent={true}
                    visible={this.state.modalVisible}
                    onRequestClose={() => {
                        this._setModalVisible(false)
                    }}>
                    <View style={Styles.container}>
                        <Text>Modal!!!</Text>
                    </View>
                </Modal>
            </View>
        )
    }
}

const Styles = StyleSheet.create({
    wrap: {
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'flex-start',
        alignItems: 'center',
    },
    container: {
        position: 'absolute',
        top: 0,
        left: 0,
        right: 0,
        bottom: 0,
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: 'rgba(0,0,0,0.7)'
    },
});

export default TestPage;
```
environment:
react native version: 0.40
android version: 7.1.1
the gif explain everything, and this phenomenon confuse me whether should i continue to use Modal component.

**Comment by walkerXiong** [comment]
any comment, any response will be appreciate.

**Comment by prtkkmrsngh** [comment]
With me flickering issue is with android < 5.0

**Comment by walkerXiong** [comment]
fixed in RN 0.45

**Comment by walkerXiong** [comment]
weird, confuse me:
=================
brands: one plus 3 
android version: 7.1.1
Modal used in hybrid application: flicker
Modal used in pure react-native app: not flicker
=================
brands: oppo
android version: 5.1
Modal used in hybrid application: not flicker
Modal used in pure react-native app: not flicker
=================
brands: red Mi
android version: 4.4.4
Modal used in hybrid application: not flicker
Modal used in pure react-native app: not flicker

so, i don't know, why one plus 3 behaving different between hybrid application and pure react native app.
